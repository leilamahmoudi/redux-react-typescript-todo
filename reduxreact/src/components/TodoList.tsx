import React, { useState } from "react";
import { useSelector } from "react-redux";
import AddTodoForm from "./AddTodoForm";
import TodoItem from "./TodoItem";
import "../styles/todoList.css";
import { store } from "../store/todos";

export type RootState = ReturnType<typeof store.getState>;
export interface Todo {
  id: string;
  title: string;
  description: string;
  isDone: boolean;
}

const TodoList: React.FC = () => {
  const todos = useSelector<RootState, Todo[]>((state) => state.todos);

  const unDoneTodos = todos.filter((item) => !item.isDone);
  const doneTodos = todos.filter((item) => item.isDone);

  const setTodos = () => {};

  return (
    <section className="todo__list">
      <h2 className="todo__list-title">What do you need to do today?</h2>
      <AddTodoForm />
      <ul id="todoList" className="list--container">
        {unDoneTodos.map((todo) => {
          return <TodoItem todo={todo} key={todo.id} />;
        })}
        {doneTodos.map((todo) => {
          return <TodoItem todo={todo} key={todo.id} />;
        })}
      </ul>
    </section>
  );
};

export default TodoList;
