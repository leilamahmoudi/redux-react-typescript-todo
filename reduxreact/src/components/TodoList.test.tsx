import React from "react";
import { render, screen } from "@testing-library/react";
import TodoList from "./TodoList";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore([]);
const store = mockStore({ todos: [] });

test("renders todo list", () => {
  render(
    <Provider store={store}>
      <TodoList />
    </Provider>
  );
  const todoListElement = screen.getByText(/what do you need to do today?/i);
  expect(todoListElement).toBeInTheDocument();
});

test("renders correctly", () => {
  const tree = render(
    <Provider store={store}>
      <TodoList />
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});
