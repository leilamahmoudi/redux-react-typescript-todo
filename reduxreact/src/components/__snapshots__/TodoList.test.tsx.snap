// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="todo__list"
      >
        <h2
          class="todo__list-title"
        >
          What do you need to do today?
        </h2>
        <div>
          <form
            class="form__container"
            id="form"
          >
            
            <input
              class="form__input-title"
              id="txtTodoItemToAdd"
              placeholder="Add new task"
              type="text"
              value=""
            />
            <input
              class="form__input-description"
              id="inputDescription"
              placeholder="description"
              type="text"
              value=""
            />
            <input
              class="form__input-add"
              id="btnAddTodo"
              type="submit"
              value="Add todo"
            />
          </form>
        </div>
        <ul
          class="list--container"
          id="todoList"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="todo__list"
    >
      <h2
        class="todo__list-title"
      >
        What do you need to do today?
      </h2>
      <div>
        <form
          class="form__container"
          id="form"
        >
          
          <input
            class="form__input-title"
            id="txtTodoItemToAdd"
            placeholder="Add new task"
            type="text"
            value=""
          />
          <input
            class="form__input-description"
            id="inputDescription"
            placeholder="description"
            type="text"
            value=""
          />
          <input
            class="form__input-add"
            id="btnAddTodo"
            type="submit"
            value="Add todo"
          />
        </form>
      </div>
      <ul
        class="list--container"
        id="todoList"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
